src/infer_request.cc

```diff
@@ -241,6 +241,13 @@ InferenceRequest::AddParameter(const char* name, const bool value)
   return Status::Success;
 }
 
+Status
+InferenceRequest::AddParameter(const InferenceParameter& parameter)
```

Comment:
One other option would be a 'SetParameters()' method that takes in a deque and uses the assignment oeprator of the dequeue.